gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Table.java

```diff
@@ -322,14 +322,15 @@ public Builder toBuilder() {
   }
 
   @Override
-  public boolean equals(Object obj) {
-    return obj instanceof Table
+  public final boolean equals(Object obj) {
+    return this == obj
+        || obj instanceof Table
```

Comment:
The method is `final` so even if extended that should not be a problem.
However I believe that in all "functional" classes we used `instanceof` while in all `Info` objects we used `obj.getClass().equals(XxxInfo.class)` as those classes are extended. I need to double check that this is consistent.